_ingest-pipelines/processors/pipeline.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: default
+title: Pipeline
+parent: Ingest processors
+nav_order: 200
+---
+
+# Pipeline processor
+
+The `pipeline` processor defines a series of processing steps to be executed on documents during the ingestion or update process. This is particularly useful when you need to perform complex data transformations or apply multiple processing steps to your data before it is indexed.
+
+The following is the syntax for the `pipeline` processor:
+
+```json
+{
+  "pipeline": {
+    "name": "<pipeline_name>",
+    "description": "<pipeline_description>",
+    "processors": [
+      {
+        "<processor_type>": {
+          "<processor_config>": "<processor_value>"
+        }
+      },
+      {
+        "<processor_type>": {
+          "<processor_config>": "<processor_value>"
+        }
+      }
+    ]
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `pipeline` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`name` | Required	| The name of the pipeline to execute.
+`processors` | Required | An array of other processor configurations to execute in the specified order.
+`ignore_missing_pipeline` | Optional | Determines whether to ignore a missing or undefined pipeline. Default is `false`.
+`description` | Optional | Description of the processor's purpose or configuration.
+`if` | Optional | Conditionally execute the processor.
+`ignore_failure` | Optional | Ignore failures for the processor. See [Handling pipeline failures]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
+`on_failure` | Optional | Handle failures for the processor. See [Handling pipeline failures]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
+`tag` | Optional | Identifier for the processor. Useful for debugging and metrics.
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+### Step 1: Create a pipeline
+
+The following query creates a pipeline named `my_pipeline` that uses the `pipeline` processor to add a new field `my_field` with a static value and convert the `message` field to uppercase: 
+
+```json
+PUT _ingest/pipeline/my_pipeline
```

Comment:
The following query creates a general pipeline named `general-pipeline`, and then creates a new pipeline named `outer-pipeline` which references the `general-pipeline`.
Create a general pipeline:
```
PUT _ingest/pipeline/general_pipeline
{
  "description": "a general pipeline",
  "processors": [
    {
      "uppercase": {
        "field": "protocol"
      },
      "remove": {
        "field": "name"
      }
    }
  ]
}
```
Create an outer pipeline:
```
PUT _ingest/pipeline/outer-pipeline
{
  "description": "an outer pipeline referencing the general pipeline",
  "processors": [
    {
      "pipeline": {
        "name": "general-pipeline"
      }
    }
  ]
}
```